testbed/src/test/java/org/spdx/testbed/util/TestCaseFinderTest.java

```diff
@@ -0,0 +1,68 @@
+package org.spdx.testbed.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.spdx.testbed.TestCase;
+import org.spdx.testbed.TestCaseCategory;
+import org.spdx.testbed.generationTestCases.GenerationTestCase;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class TestCaseFinderTest {
+    private static final String GENERATION_MINIMAL = "generationMinimalTest";
+    private static final String GENERATION_BASELINE_SBOM = "generationBaselineSbomTest";
+    private static final String GENERATION_DOCUMENT = "generationDocumentTest";
+    private static final String GENERATION_PACKAGE = "generationPackageTest";
+    private static final String GENERATION_FILE = "generationFileTest";
+    private static final String GENERATION_SNIPPET = "generationSnippetTest";
+    private static final String GENERATION_LICENSE = "generationLicenseTest";
+    private static final String GENERATION_RELATIONSHIP = "generationRelationshipTest";
+    private static final List<String> generationTestCaseNames = List.of(GENERATION_MINIMAL,
+            GENERATION_BASELINE_SBOM, GENERATION_DOCUMENT, GENERATION_PACKAGE, GENERATION_FILE,
+            GENERATION_SNIPPET, GENERATION_LICENSE, GENERATION_RELATIONSHIP);
+    private TestCaseFinder testCaseFinder;
+
+    @BeforeEach
+    public void setup() {
+        testCaseFinder = new TestCaseFinder();
+    }
+
+
+    @Test
+    public void findGenerationTestCases() {
+        var testCases =
+                testCaseFinder.findTestCasesByCategories(List.of(TestCaseCategory.GENERATION));
+
+        assertThat(testCases).allMatch(element -> element instanceof GenerationTestCase);
+        var testCaseNames = testCases.stream().map(TestCase::getName).collect(Collectors.toList());
+        assertThat(testCaseNames).containsExactlyInAnyOrderElementsOf(generationTestCaseNames);
```

Comment:
At this stage, no. The `TestCaseFinder` doesn't care about the ordering. The `TestCaseSelector` orders alphabetically in case no names are specified. I will have a look if a test covers that functionality.

Reply:
Added the ordering aspect to a test for `TestCaseSelector`